script.js

```diff
@@ -1,12 +1,207 @@
+    option.value = show.name;
+    option.textContent = show.name;
+    selectShows.appendChild(option);
+  });
+}
+
+// Search specific Shows based on select
+async function selectShow(event) {
+  if (event.target.value == "allSerials") {
+    setup();
+    //Remove options from episodes's select
+    const options = selectEpisodes.getElementsByTagName("option");
+    for (let i = options.length - 1; i > 0; i--) {
+      selectEpisodes.removeChild(options[i]);
+    }
+    searchUser.value = "";
+    searchUser.disabled = true;
+  } else {
+    searchUser.value = "";
+    searchUser.disabled = false;
+    //Remove options from episodes's select
+    const options = selectEpisodes.getElementsByTagName("option");
+    for (let i = options.length - 1; i > 0; i--) {
+      selectEpisodes.removeChild(options[i]);
+    }
+    // Find selected movie
+    const specificShow = allShows.find(
+      (show) => show.name === event.target.value
+    );
+    // Determine show's id
+    const specificShowID = specificShow.id;
+    // Get all show's episodes
+    allEpisodes = await getAllAvailableEpisodes(specificShowID);
+    // make card for each episode
+    makePageForEpisodes(allEpisodes);
+  }
+}
+
+//Get all episodes via fetch
+async function getAllAvailableEpisodes(showId) {
+  // await response of the fetch call
+  const response = await fetch(
+    `https://api.tvmaze.com/shows/${showId}/episodes`
+  );
+  // only proceed once its resolved
+  const data = await response.json();
+  // only proceed once second promise is resolved
+  return data;
+}
+
+//make card for each episode
 function makePageForEpisodes(episodeList) {
-  const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+  let output = "";
+  episodeList.forEach((episode) => {
+    output += `
+      <div class="col">
+        <div class="card h-100 px-0 mx-1">
+        <img src="${episode.image.medium}" class="card-img-top mx-0" alt="...">
+        <div class="card-body mb-4">
+        <h5 >${episode.name} - S${episode.season
+      .toString()
+      .padStart(2, "0")}E${episode.season.toString().padStart(2, "0")}</h5>
+      <p>${
+        episode.summary.split(" ").length >= 25
+          ? episode.summary.split(" ").slice(0, 24).join(" ").concat("...")
+          : episode.summary.split(" ").slice(0, 24).join(" ")
+      }</p>
+        </div>
+        <a href="${
+          episode.url
+        }" class="btn btn-primary" style="position:absolute; bottom:10px; left:20px; right:20px;" target="_blank">See More</a>
+      </div>
+      </div>
+    `;
+    showEpisodseRow.innerHTML = output;
+  });
+
+  // Create option for each episodes in select after loading page
+  episodeList.forEach((episode) => {
+    const option = document.createElement("option");
+    option.value = `S${episode.season
+      .toString()
+      .padStart(2, "0")}E${episode.number.toString().padStart(2, "0")}`;
```

Comment:
this logic works well, but again is hard to debug as it is in a template string. Can you think of a way to reuse this logic for the output string above (previous comment), so that it is correct in both places?